src/ceph-volume/ceph_volume/devices/raw/list.py

```diff
@@ -27,64 +49,89 @@ def __init__(self, argv):
+        #           },
+        #           {"name": "/dev/sdd", "maj:min": "8:48", "rm": "0", "size": "1T", "ro": "0", "type": "disk", "mountpoint": null,
+        #               "children": [
+        #                   {"name": "/dev/mapper/ceph--a26fd8ef--bd39--46f1--b21e--a1596c6d74de-osd--block--8a5c79e2--448e--45ff--bf7c--d2d1dddd5db2", "maj:min": "253:0", "rm": "0", "size": "512G", "ro": "0", "type": "lvm", "mountpoint": null},
+        #                   {"name": "/dev/mapper/ceph--a26fd8ef--bd39--46f1--b21e--a1596c6d74de-osd--block--be039c3c--b0a4--47cf--a731--b07755d71fe4", "maj:min": "253:1", "rm": "0", "size": "512G", "ro": "0", "type": "lvm", "mountpoint": null}
+        #               ]
+        #           }
+        #       ]
+        #   }
+        lsblk_command = ['lsblk', '--paths', '--json']
+        if devs is not None:
+            for dev in devs:
+                lsblk_command.append(dev)
+        out, err, ret = process.call(lsblk_command)
+        lsblk_json = json.loads(''.join(out))
         result = {}
-        for dev in devs:
+        for dev_info in lsblk_json['blockdevices']:
+            dev = dev_info['name']
             logger.debug('Examining %s' % dev)
-            # bluestore?
-            out, err, ret = process.call([
-                'ceph-bluestore-tool', 'show-label',
-                '--dev', dev], verbose_on_failure=False)
-            if ret:
-                logger.debug('No label on %s' % dev)
-                continue
-            oj = json.loads(''.join(out))
-            if dev not in oj:
-                continue
-            if oj[dev]['description'] != 'main':
-                # ignore non-main devices, for now
+
+            # Because children could be phantom Atari partitions mentioned above, first check if the
+            # parent is a valid bluestore OSD. If it is, the children will only exist if they are
+            # phantom Atari partitions and should be ignored. If it isn't, then the children could
+            # be individual bluestore OSDs.
+            is_bluestore = False
+            try:
+                if disk.has_bluestore_label(dev):
+                    is_bluestore = True
+            except OSError as e:
+                logger.error(('failed to determine if device {} is BlueStore.'.format(dev),
+                             'device children will not be examined to avoid reporting invalid BlueStore data from phantom Atari partitions.',
+                             'err: {}'.format(e)))
                 continue
-            whoami = oj[dev]['whoami']
-            result[oj[dev]['osd_uuid']] = {
-                'type': 'bluestore',
-                'osd_id': int(whoami),
-                'osd_uuid': oj[dev]['osd_uuid'],
-                'ceph_fsid': oj[dev]['ceph_fsid'],
-                'device': dev
-            }
+
+            if is_bluestore:
+                bs_info = _get_bluestore_info(dev)
+                if bs_info is not None:
```

Comment:
Avoid the negative condition?

```python
if bs_info is None:
  continue
result[bs_info['osd_uuid']] = bs_info
```

Reply:
What if `bs_info` isn't that an error if we populate nothing?

Reply:
Good idea

Reply:
Related to your other comment, I think yes. It is an error here since we **know** the disk has a bluestore header. But for the later case where we are looking at children, we don't check for the bluestore header and assume no info means it's not bluestore...

I guess it's a little strange to not check in both places, but the try-except is a little verbose.